packages/tools/devtools/devtools-view/src/components/Menu.tsx

```diff
@@ -314,6 +340,19 @@ export interface MenuItemProps {
 	onClick: (event: unknown) => void;
 	text: string;
 	isActive: boolean;
+	/**
+	 * Icon to display next to the container name based on its state.
+	 *
+	 * @defaultValue `undefined` - No state icon is displayed when not provided.
+	 */
+	readonly stateIcon?: React.ReactElement;
```

Comment:
If we're going to show an icon for the disconnected / closed states, we might as well show an icon for all states IMO. That will help make the layout a bit more consistent too.

Reply:
Made some huge changes here. And thanks for the great catch here, I've realized we were rendering same state icon for disconnected & readonly states (in `IContainerRuntime`). I had to make some changes in the `devtools-core` package. 

Would like to have your thoughts. 


Also, I'm thinking it'd be more intuitive to not have any icon when the container is connected? It takes a closer to look distinguish the connected / disconnected state (or we can find a better icon). 
<img width="290" height="297" alt="Screenshot 2025-08-15 at 18 59 28" src="https://github.com/user-attachments/assets/1c113376-f185-45f9-aa5b-553e1d6b85fd" />